src/neural/backends/network_onnx.cc

```diff
@@ -457,11 +460,25 @@ OnnxNetwork::OnnxNetwork(const WeightsFile& file, const OptionsDict& opts,
   std::transform(outputs_.begin(), outputs_.end(),
                  std::back_inserter(outputs_cstr_),
                  [](const auto& x) { return x.c_str(); });
+  uint64_t hash = 0;
+  if (provider == OnnxProvider::TRT) {
+    auto sv = md.model();
+    // Filter out 16 random chars in the name of our conversions.
```

Comment:
Maybe let's just remove that random name from the converter? Or even not fill that model at all.

Alternatively, we can populate it with some stable hash while converting. Like maybe hash of the source weights file.

Also, it seems that domain+name is something that could be use as a key rather than ONNX file hash:
https://github.com/onnx/onnx/blob/c25eebcf51b781dbfcc75a9c8bdf5dd1781367fe/onnx/onnx.proto#L471-L473

(and also it seems that `org.lczero` part should go to domain?)

Reply:
I went with removing the random name. When we move to the new backend interface I'll change it to the network name.